actors/runtime/src/runtime/fvm.rs

```diff
@@ -91,10 +91,60 @@ impl MessageInfo for FvmMessage {
     }
 }
 
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    // SysErrIllegalActor
+    #[error("Method must validate caller identity exactly once")]
+    CallerAlreadyValidated,
+    // SysErrForbidden
+    #[error("caller {0} ist not one of supported")]
+    UnsupportedCallerAddress(Address),
+    // SysErrForbidden
+    #[error("caller cid type {0} not one of supported")]
+    UnsupportedCallerType(Cid),
+    #[error("syscall: {0}")]
+    Syscall(#[from] ErrorNumber),
+    // ErrIllegalState
+    #[error("failed to create state; expected empty array CID, got: {0}")]
+    RootNotEmpty(Cid),
+    #[error("encoding: {0}")]
+    CborStore(anyhow::Error),
+    #[error("no state: {0}")]
+    NoState(#[from] NoStateError),
+    #[error("exit code: {0}")]
+    NonZeroExitCode(ExitCode),
+    // SysErrIllegalActor
+    #[error("runtime.send() is not allowed")]
+    SendNotAllowed,
+    #[error("actor delete: {0}")]
+    ActorDelete(#[from] fvm::error::ActorDeleteError),
+}
```

Comment:
We should define errors for specific syscalls.

Reply:
Otherwise the user has to deal with a bunch of non-specific errors.

Reply:
Could you expand on this a little, do you mean having one error variant per syscall error, or just a general `Syscall(SyscallError)` variant, or sth else?

Reply:
Ah, sorry. I mean different error types for different runtime functions.

Reply:
I concur